### PR TITLE
wip: notifications and start architecture for sql schema

### PR DESCRIPTION
**Pull request by gbibeaul**
TL/DR:

- incorporated a central table to generate analytics views
- work on an event system
- each ticket is now saved in the DB
- Notification UI
Supbase Schema-1
 (WIP)
Screen Shot 2021-12-26 at 10 08 39 PM

### CODE REVIEW

File: apps/gamble/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const NOTIFICATION_APPROVAL = 'I approve Gamblefi.xyz notifying me of important events'
+
+/**
+ * matches the database
+ */
+export enum NetworkIds {
+    Local= 1,
+    Fuji = 1,
```
Comment (ethersage) [inline apps/gamble/src/constants.ts:8]: Is using the same ID for`Local` and `Fuji` intentional?

File: apps/gamble/src/routes/api/send-notification.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+		if (typeof originatingAffress !== 'string') {
+			throw 'originating email address not set in env vars';
+		}
+
+		const apiKey = await supabase.from('api_keys').select('value').eq('id', 1).limit(1)
+		
+		if(!isBearerValid(request.headers.authorization, apiKey.data[0])) {
+			return {
+				message: 'not authorized'
```
Comment (ethersage) [inline apps/gamble/src/routes/api/send-notification.ts:20]: Does this need a `status` too?

File: apps/gamble/src/routes/api/send-notification.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			from: String(import.meta.env.VITE_GMAIL_ADDRESS),
+			to: email,
+			subject,
+			text
+		};
+
+		emailProvider.sendMail(mailOptions);
+
+		return {
+			sucess: true
```
Comment (ethersage) [inline apps/gamble/src/routes/api/send-notification.ts:36]: Mispelled `success`

File: apps/gamble/src/routes/api/send-notification.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		};
+	} catch (error) {
+		console.error(error);
+		return {
+			error: JSON.stringify(error)
```
Comment (ethersage) [inline apps/gamble/src/routes/api/send-notification.ts:41]: Are you potentially accidentally revealing privileged backend info with this?

File: apps/gamble/src/routes/lotto.svelte [hunk 5]
```diff
@@ -11,22 +11,27 @@
...
+	import NotificationPannel from '../components/lotto/NotificationPannel.svelte';
+
 </script>
 
 <div class="h-full gradient justify-content-around">
 	<Banner />
 	<div class="lg:flex lg:justify-between px-2">
 		<TicketPurchase />
 		<TicketReview />
+		<NotificationPannel />
```
Comment (ethersage) [inline apps/gamble/src/routes/lotto.svelte:35]: Panel only has one `n` And I am a glorified spell check

File: apps/gamble/src/stores/lotto.ts [hunk 6]
```diff
@@ -62,21 +71,49 @@ const createLotto = () => {
...
+				currency: initialState.currency
+			};
+
+			fetch('/api/save-ticket', {
+				method: 'POST',
+				body: JSON.stringify({
+					transaction_id: tx.transactionHash,
+					numbers: betsToBigNumbers,
+					wallet_id: walletAddress
+				})
```
Comment (ethersage) [inline apps/gamble/src/stores/lotto.ts:109]: Where’s the actual blockchain tx?

File: packages/LottoWinnerMock/LottoWinnerMock.dbg.json [hunk 7]
```diff
@@ -1,4 +1,4 @@
 {
   "_format": "hh-sol-dbg-1",
-  "buildInfo": "../../build-info/face00878df7aa20fcd5141dd1135697.json"
+  "buildInfo": "../../build-info/d77b4d97bcfc9c0bbbf07e1e332b4261.json"
 }
```
Comment (ethersage) [inline packages/LottoWinnerMock/LottoWinnerMock.dbg.json:4]: Should this file be getting checked in?

**Review by ethersage — APPROVED** [review]

File: apps/gamble/src/constants.ts (hunk 1 above)
Comment (gbibeaul) [reply apps/gamble/src/constants.ts:8]: in this case yes! I want to record in the DB any test event as 1.

**Pull request merged**